### PR TITLE
Clarify ignore_above behavior with arrays of strings

### PR DESCRIPTION
**Pull request by dliappis**
and also note that all string(s) will still be visible in the
`_source` field.

### CODE REVIEW

File: docs/reference/mapping/params/ignore-above.asciidoc [hunk 1]
```diff
@@ -2,6 +2,9 @@
 === `ignore_above`
 
 Strings longer than the `ignore_above` setting will not be indexed or stored.
+For arrays of strings, `ignore_above` will be applied for each array element and string elements longer than `ignore_above` will not be indexed or stored.
```
Comment (colings86) [inline docs/reference/mapping/params/ignore-above.asciidoc]: minor: maybe `...each array element separately and...` would be clearer here?

**Review by colings86 — APPROVED** [review]
LGTM, I left a minor comment but I don't feel strongly about it so happy for this to be merged with or without my suggestion applied

File: docs/reference/mapping/params/ignore-above.asciidoc (hunk 1 above)
Comment (dliappis) [reply docs/reference/mapping/params/ignore-above.asciidoc]: Will fix, thanks!

Comment (dliappis) [reply docs/reference/mapping/params/ignore-above.asciidoc]: Fixed in https://github.com/elastic/elasticsearch/pull/33057/commits/eac5614eef0c4b527a94ad19f93103134848da0d as per your suggestion, I think it looks clearer now WDYT?

**Review by colings86 — APPROVED** [review]
LGTM

**Comment by dliappis** [comment]
Thanks a lot @colings86 for the blazingly fast review ; thoughts about how far back to go for the backport after CI is green?

**Comment by colings86** [comment]
I think just back to 6.3 is fine, since these are the active versions

**Comment by dliappis** [comment]
Cherry-picked:

- 6.x: 14be9d7d28685cec64538b22b7be979a5ac7392a
- 6.4: 86393d5d4e4ae5b159c25b6abe7a78bcc398c3fb
- 6.3: 34d7dbca2356600bf65d983a450f1db0c5b18444

**Pull request merged**